src/components/panel-tabs/with-memory.js

```diff
@@ -29,7 +29,7 @@ const withMemory = WrappedComponent => {
 
 		const initialTab = props.initialTab || // If there's a prop, use it.
 			tabCache[ props.blockProps.clientId ] || // Or check if there's a previously selected tab.
-			( isUnmodifiedBlock( props.blockProps ) ? 'layout' : 'style' ) // Or if not, default to layout or the style tab.
+			( isUnmodifiedBlock( props.blockProps ) ? ( props.tabs || [ 'layout', 'style', 'advanced' ] )[ 0 ] : 'style' ) // Or if not, default to layout or the style tab.
```

Comment:
Maybe we can use lodash. Untested code:
```suggestion
			( isUnmodifiedBlock( props.blockProps ) ? ( first( props.tabs ) || 'layout' ) : 'style' ) // Or if not, default to layout or the style tab.
```